### ISSUE TITLE
Cannot build tests after adding v3.1 in my project

### ISSUE THREAD

**Issue by kudoleh** [forcedotcom/SalesforceMobileSDK-iOS#962]
After adding 3.1 release to an existing project using cocoapods, I cannot run the tests anymore. They fail to build.
The compiler complains about not finding some headers, like

#import <SalesforceOAuth/SFOAuthCredentials.h> inside SFRestRequest.h

I have also tried not with cocoapods, but by creating a sample project using forceios create command, and the problem persists. I created a repository with this sample project so that the issue is easy to reproduce: https://github.com/kudoleh/SalesforceMobileSDK-iOS

**Comment by kudoleh** [comment]
I managed to fix this problem by replacing all imports in.h files of SaleForce framework:
#import <SalesforceOAuth/SFOAuthCoordinator.h>
#import <SalesforceOAuth/SFOAuthInfo.h>
TO 
#import "SFOAuthCoordinator.h"
#import "SFOAuthInfo.h"

All afected files are: SFRestAPI.h, SFRestRequest.h, SFAuthenticationManager.h, SFUserAccount.h, SFUserAccountManager.h

However I could not do the pull request. This changes does not work in project of saleforce examples project.

**Comment by kudoleh** [comment]
I found out another better way to fix it without modifying the framework.
In the Header Search Paths of test target I put  "${PODS_ROOT}/Headers" with RECURSIVE option.
This fixes test build. 

This is strange that I had to do it only for SalesForce framework, because all another frameworks that I use does not need this change.

**Comment by khawkins** [comment]
Which tests are you trying to run, the tests of your app project?  It looks like you're running into Header path issues.  For all of our subspecs, we expect consuming projects to reference e.g. `${PODS_ROOT}/Headers/Public/SmartSync/Headers` in `HEADER_SEARCH_PATHS`, to allow for setting imports to the umbrella folder of each subspec.  This configuration is added via the `xcconfig` property of each subspec, which CocoaPods seems to be able to set for the main app target.  But if you're creating other targets (like a test target), looks like you'll have to set those paths yourself.

[This StackOverflow post](http://stackoverflow.com/questions/14512792/libraries-not-found-when-using-cocoapods-with-ios-logic-tests) looks like it's on the right track for using the pod xcconfig in your test target as well.

**Comment by kudoleh** [comment]
Yes, thanks, I updated my fix in HEADER_SEARCH_PATHS from 
 "${PODS_ROOT}/Headers" recursive 
 to
"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers" not-recursive 
and it works.

**Comment by kudoleh** [comment]
The strange thing that I have all these headers coming with $inherited in test target like:
"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers/SaleforceRestAPI"
"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers/SaleforceRestAPI"
"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers/SmartSync" ....

Then to make it work why I have to import "${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers" non-recursive, they are already imported.

So in conclusion if you create the project using forceios create and then create test target, you need to copy paste the HEADER_SEARCH_PATHS from main target.

And if you use cocoapods to install the framework on your existing project then "${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers" is needed in HEADER_SEARCH_PATHS test target.

**Comment by khawkins** [comment]
In the non-recursive example, e.g. `"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers/SaleforceRestAPI"` is not going to point to the right place, for umbrella imports anyway, because that's the folder path to the actual header files.  I.e. you could `#import <SFRestAPI.h>` from there, but not `#import <SalesforceRestAPI/SFRestAPI.h>`.  Your header search path should be up a level from there (`"${PODS_ROOT}/Headers/Public/SalesforceMobileSDK-iOS/Headers"`), to make the umbrella header imports work.  The good thing is, as you mentioned at the end, there's only the one header path to add for all of the subspecs.
